index.js

```diff
@@ -7,36 +7,73 @@ class MailchimpSubscribe {
     this.baseUrl = `https://${apiKey.split('-')[1]}.api.mailchimp.com/3.0`;
   }
 
-  request(endpoint, method, data, done) {
-    wreck.request(method, `${this.baseUrl}${endpoint}`, {
+  async request(endpoint, method, data) {
+    const response = await wreck.request(method, `${this.baseUrl}${endpoint}`, {
       headers: {
         Authorization: `Basic ${new Buffer(`mailchimp_user:${this.apiKey}`).toString('base64')}`
       },
       payload: JSON.stringify(data)
-    }, (err, response, json) => {
-      if (err) {
-        return done(err);
-      }
-      wreck.read(response, { json: true }, (err, body) => {
-        if (response.statusCode !== 200) {
-          return done(body);
-        }
-        done(null, body);
-      });
     });
+
+    const body = await wreck.read(response, { json: true });
+    if (response.statusCode !== 200) {
+      throw new Error(body.detail);
+    }
+
+    return body;
+  }
+
+  listInterestCategories(listId) {
+    return this.request(`/lists/${listId}/interest-categories`, 'GET', {});
+  }
+
+  interestCategoryInfo(listId, interestCategoryId) {
+    return this.request(`/lists/${listId}/interest-categories/${interestCategoryId}`, 'GET', {});
+  }
+
+  listInterestsByCategory(listId, interestCategoryId) {
+    return this.request(`/lists/${listId}/interest-categories/${interestCategoryId}/interests`, 'GET', {});
   }
 
-  listInterests(listId, done) {
-    this.request(`/lists/${listId}/interest-categories`, 'GET', {}, done);
+  async listAllInterests(listId) {
+    const categories = await this.listInterestCategories(listId);
```

Comment:
lets do some sort of memory caching here so we aren't re-fetching every subscribe